### PR TITLE
M3-3228 Add warning when user creates a new snapshot

### PR DESCRIPTION
**Pull request by WilkinsKa1**
## Description

For manual snapshots, we will now present a dialog to allow the user to confirm/cancel that they want to proceed with the new snapshot and inform them they will lose their previous snapshot. 

## Type of Change
- Non breaking change ('update')

## Note to Reviewers

Please test creating various snapshots and verifying errors (one way is to try creating a snapshot while one is already in progress)

Also, does this make sense as a dialog? The requested copy from classic seems to fit our dialog pattern, but maybe a warning notice is all that is needed. Let me know what you think!

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/DestructiveSnapshotDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    const title = 'Take a snapshot?';
+
+    return (
+      <ConfirmationDialog
+        open={this.props.open}
+        title={`${title}`}
+        onClose={this.props.onClose}
+        actions={this.renderActions}
+      >
+        {error && <Notice error text={error} />}
```
Comment (martinmckenna) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/DestructiveSnapshotDialog.tsx]: The `ConfirmationDialog` component takes an `error` prop. I'd just pass this directly to it.

File: packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx [hunk 2]
```diff
@@ -166,6 +168,11 @@ interface State {
     backupCreated: string;
     backupID?: number;
   };
+  destructiveDialog: {
+    open: boolean;
+    mode: 'snapshot';
+    error?: string;
+  };
```
Comment (martinmckenna) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: It's a bit of a bad pattern to nest state like this. (This pattern is elsewhere in the app I know, but _should_ be refactored).

I would instead just make your state flat and give them names like `dialogOpen`, `dialogError`, etc.

Also, you definitely are going to want to add some loading state here too. Set it to true before the request fires and set it to false when it succeeds or finishes.

Comment (martinmckenna) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: Also not sure if you were planning on using `mode` for something else, but it seems mostly unnecessary with this logic.

Comment (WilkinsKa1) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: Where and what kind of loading are you thinking? When initiating a manual backup we get the progress loader + loading spinner.. do you mean in between the time it takes for the toast to display and then the progress loader appearing?

Comment (martinmckenna) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: just a loading indicator on the submit button after the user has clicked submit but before the request succeeds or fails.

Check out `DeleteFirewallDialog.tsx` as an example. The flow would basically go like this

```js

class MyComponent extends React.Component {
  state: {
    loading: false,
    error: undefined,
  }

  doSomeRequest = () => {
    this.setState({ loading: true })

    makeAPIRequest()
      .then(() => {
        this.setState({ loading: false })
        closeDialog();
      })
      .catch(e => {
        this.setState({
          error: e,
          loading: false
        })
      })

    render() {
      return (
        <Dialog error={this.state.error}>
          <Button loading={this.state.loading}>Submit</Button>
        </Dialog>
      )
    }
  }
}
```

Comment (WilkinsKa1) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeBackup/LinodeBackup.tsx]: Thanks for the explanation + example! I updated, so all your feedback should be addressed now whenever you get a chance to re-review.

**Pull request merged**